lib/ProductOpener/Display.pm

```diff
@@ -10272,6 +10212,285 @@ CSS
 	return $template_data_ref;
 }
 
+=head2 should_show_nutrient ( $nutriment, $product_ref, $request_ref )
+
+Determines if a specific nutrient should be displayed in the nutrition table.
+
+This function checks various conditions such as whether the nutrient has a value, is optional, or is country-specific, and returns a boolean indicating whether the nutrient should be shown.
+
+=head3 Arguments
+
+=head4 Nutrient $nutriment
+
+The nutrient identifier (e.g., "fat", "sodium", etc.).
+
+=head4 Product reference $product_ref
+
+Reference to the product's data, which contains the nutritional values.
+
+=head4 Request reference $request_ref
+
+Reference to request data, which can contain country-specific or other relevant information.
+
+=head3 Return values
+
+Returns 1 if the nutrient should be shown, 0 otherwise.
+
+=cut
+
+sub should_show_nutrient {
+	my ($nutriment, $product_ref, $request_ref, $nid) = @_;
+
+	# Logic to determine if the nutrient should be shown (as in your original code)
+	my $is_nutrient_with_value = (
+		((defined $product_ref->{nutriments}{$nid}) and ($product_ref->{nutriments}{$nid} ne ''))
+			or ((defined $product_ref->{nutriments}{$nid . "_100g"})
+			and ($product_ref->{nutriments}{$nid . "_100g"} ne ''))
+			or ((defined $product_ref->{nutriments}{$nid . "_prepared"})
+			and ($product_ref->{nutriments}{$nid . "_prepared"} ne ''))
+			or ((defined $product_ref->{nutriments}{$nid . "_modifier"})
+			and ($product_ref->{nutriments}{$nid . "_modifier"} eq '-'))
+			or ((defined $product_ref->{nutriments}{$nid . "_prepared_modifier"})
+			and ($product_ref->{nutriments}{$nid . "_prepared_modifier"} eq '-'))
+	);
+
+	# Logic to show important or non-optional nutrients
+	if (($nutriment !~ /-$/) or $is_nutrient_with_value) {
+		return 1;
+	}
+
+	# Hide non-important nutrients when no value is present
+	if ((($nutriment !~ /^!/) or ($product_ref->{id} eq 'search')) and not($is_nutrient_with_value)) {
+		return 0;
+	}
+
+	# Special handling for nutrition scores (optional, based on request_ref)
+	if ($nid =~ /^nutrition-score-(.*)$/) {
+		# Always show FR score and Nutri-Score
+		if (($request_ref->{cc} ne $1) and (not($1 eq 'fr'))) {
+			return 0;
+		}
+
+		# Experimental: hide nutrition scores for now
```

Comment:
Can we add a hyperlink to help decide if+when these should be displayed in future?